core/src/main/java/org/apache/calcite/sql/SqlUtil.java

```diff
@@ -305,7 +305,13 @@ public static void unparseFunctionSyntax(SqlOperator operator,
       }
       SqlIdentifier id = function.getSqlIdentifier();
       if (id == null) {
-        writer.keyword(operator.getName());
+        // The following code block is executed exclusively when the code flow originates from mig.
+        if (((SqlFunction) operator).getFunctionType() == SqlFunctionCategory.USER_DEFINED_FUNCTION
```

Comment:
create a method and move this code .

private boolean shouldUnparseAsLowerCase(SqlFunction function, SqlWriter writer) {
    return function.getFunctionType() == SqlFunctionCategory.USER_DEFINED_FUNCTION &&
           writer.isUDFLowerCase();
}


also further convert code to ternary 

shouldUnparseAsLowerCase(function, writer)
        ? operator.getName().toLowerCase()
        : operator.getName()